assets/src/edit-story/components/dropDown/index.js

```diff
@@ -0,0 +1,305 @@
+        } else if (event.key === 'Escape') {
+          event.preventDefault();
+          setOpenOptions(!openOptions);
+          DropDown.selectRef.current.focus();
+        } else {
+          const searchTerm = searchValue + String.fromCharCode(event.keyCode);
+          setSearchValue(searchTerm);
+          clearSearchValue();
+        }
+        break;
+      default:
+    }
+  };
+
+  const clearSearchValue = useCallback(
+    debounce(800, () => {
+      setSearchValue('');
+    })
+  );
+
+  useEffect(() => {
+    if (searchValue !== '') {
+      const searchIndex = options.findIndex((item) =>
+        item.name.toLowerCase().startsWith(searchValue.toLowerCase())
+      );
+      if (searchIndex >= 0) {
+        DropDown.arrayOfOptionsRefs[searchIndex].focus();
+      }
+    }
+  }, [searchValue, options]);
+
+  useEffect(() => {
+    if (openOptions) {
+      DropDown.arrayOfOptionsRefs[0].focus();
+    }
+  }, [openOptions]);
+
+  const setOptionRef = (element) => {
+    if (element !== null) {
+      DropDown.arrayOfOptionsRefs.push(element);
+    }
+  };
+
+  const setSize = options.length;
+
+  return (
+    <DropDownContainer ref={DropDown.wrapperRef} tabIndex={-1}>
+      <DropDownSelect
+        onClick={handleOpenOptions}
+        onKeyDown={handleOpenOptions}
+        aria-pressed={openOptions}
+        aria-haspopup={true}
+        aria-expanded={openOptions}
+        disabled={disabled}
+        ref={DropDown.selectRef}
+        aria-disabled={disabled}
+      >
+        <DropDownTitle>
+          {(activeItem && activeItem.name) ||
+            __('Select an Option', 'web-stories')}
+        </DropDownTitle>
+        <DropDownIcon />
+      </DropDownSelect>
+      <DropDownListWrapper>
+        {openOptions ? (
+          <DropDownList>
+            {options.map(({ name, value: optValue }, index) => {
+              return (
+                <DropDownItem
+                  id={optValue}
+                  aria-selected={focusedOption === optValue}
+                  key={optValue}
+                  onClick={(e) => handleOptionsEvents(optValue, index, e)}
+                  onKeyDown={(e) => handleOptionsEvents(optValue, index, e)}
+                  ref={setOptionRef}
+                  aria-posinset={index}
+                  aria-setsize={setSize}
+                >
+                  {name}
+                </DropDownItem>
+              );
+            })}
+          </DropDownList>
+        ) : (
+          [clearOptionsRefs(), null]
+        )}
+      </DropDownListWrapper>
+    </DropDownContainer>
+  );
+}
+
+DropDown.propTypes = {
+  value: PropTypes.any.isRequired,
+  options: PropTypes.array.isRequired,
+  onChange: PropTypes.func.isRequired,
+  disabled: PropTypes.bool,
+};
+
+DropDown.defaultProps = {
+  disabled: false,
+};
+
+const clickOutsideConfig = {
+  handleClickOutside: () => DropDown.handleClickOutside,
+};
+
+export default onClickOutside(DropDown, clickOutsideConfig);
```

Comment:
Can you use the new `useFocusOut` hook instead of relying on this third-party package? The color picker uses that already.

Reply:
Done